src/app/hooks/useWalletReducer.ts

```diff
@@ -102,34 +105,43 @@ const useWalletReducer = () => {
     dispatch(getActiveAccountsAction(walletAccounts));
   };
 
-  const unlockWallet = async (password: string) => {
+  const migrateLegacySeedStorage = async (password: string) => {
     const pHash = await generatePasswordHash(password);
-    const decrypted = await decryptSeedPhrase(encryptedSeed, password);
+    await decryptSeedPhraseCBC(encryptedSeed, pHash.hash).then(async (decrypted) => {
+      await initSeedVault(password);
+      await storeSeed(decrypted);
+      localStorage.removeItem('salt');
+      dispatch(storeEncryptedSeedAction(''));
+      return decrypted;
+    });
+  };
+
+  const unlockWallet = async (password: string) => {
+    if (encryptedSeed && encryptedSeed.length > 0) {
+      await migrateLegacySeedStorage(password);
+      return;
+    }
+    const decrypted = await unlockVault(password);
     try {
       await loadActiveAccounts(decrypted, network, selectedNetwork, accountsList);
     } catch (err) {
       dispatch(fetchAccountAction(accountsList[0], accountsList));
       dispatch(getActiveAccountsAction(accountsList));
     } finally {
-      chrome.storage.session.set({
-        pHash: pHash.hash,
-      });
       setSessionStartTime();
     }
-    dispatch(unlockWalletAction(decrypted));
     return decrypted;
```

Comment:
Don't think we should return the seed phrase here. Opens it up to accidental in-memory storage. Rather only ever expose the seed phrase form the vault.

Reply:
updated